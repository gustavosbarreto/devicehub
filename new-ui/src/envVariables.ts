/* eslint-disable */
// import.meta.env.MODE
export const envVariables = {
  isEnterprise: process.env.VUE_APP_SHELLHUB_ENTERPRISE === "true",
  isCloud: process.env.VUE_APP_SHELLHUB_CLOUD === "true",
  stripePublishableKey: process.env.VUE_APP_SHELLHUB_STRIPE_PUBLISHABLE_KEY,
  billingEnable: process.env.VUE_APP_SHELLHUB_BILLING === "true",
  version: process.env.SHELLHUB_VERSION,
  announcementsEnable: process.env.VUE_APP_SHELLHUB_ANNOUNCEMENTS === "true",
  stripeKey: process.env.VUE_APP_SHELLHUB_STRIPE_PUBLISHABLE_KEY,
};
