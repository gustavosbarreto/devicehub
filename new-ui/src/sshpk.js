const NodeRSA = require("node-rsa");
const sshpk = require('sshpk');

const createSignatureOfPrivateKey = (
  privateKeyData,
  username
) => {
  let signature;
  const key = NodeRSA(privateKeyData);
  key.setOptions({ signingScheme: "pkcs1-sha1" });
  signature = encodeURIComponent(key.sign(username, "base64"));
  return signature;
};

const createKeyFingerprint = (privateKeyData) => {
  const key = sshpk.parsePrivateKey(privateKeyData);
  const fingerprint = key.fingerprint("md5").toString('hex');
  return fingerprint;
};

const parsePrivateKey = (privateKey) => {
  const key = sshpk.parsePrivateKey(privateKey);
  return key;
};

const convertKeyToFingerprint = (privateKey) => {
  const fingerprint = sshpk.parsePrivateKey(privateKey).fingerprint('md5');
  return fingerprint;
};

const createSignerAndUpdate = (privateKey, username) => {
  const signer = privateKey.createSign("sha512");
  signer.update(username);
  return signer.sign().toString();
};

window.global.createSignatureOfPrivateKey = createSignatureOfPrivateKey;
window.global.createKeyFingerprint = createKeyFingerprint;
window.global.convertKeyToFingerprint = convertKeyToFingerprint;
window.global.createSignerAndUpdate = createSignerAndUpdate;
window.global.parsePrivateKey = parsePrivateKey;
