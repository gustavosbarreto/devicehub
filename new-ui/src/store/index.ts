import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { mobile, MobileState } from "./modules/mobile";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { announcement, AnnouncementState } from "./modules/announcement";

export interface State {
  auth: AuthState;
  billing: NamespacesState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  mobile: MobileState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    auth,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    mobile,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
