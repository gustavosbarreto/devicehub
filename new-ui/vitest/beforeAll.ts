import { beforeAll } from "vitest";
import ResizeObserver from "resize-observer-polyfill";

(global as any).CSS = { supports: () => false };
global.ResizeObserver = ResizeObserver;

beforeAll(() => {
  global.CSS = {
    supports: (str: string) => false,
    escape: (str: string) => str,
  };
});
