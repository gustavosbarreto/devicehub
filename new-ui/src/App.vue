<template>
  <v-app>
    <component :is="layout" :data-test="layout + '-component'" />
  </v-app>
</template>

<script lang="ts">
import { defineComponent, onMounted, computed } from "vue";
import { useRouter } from "vue-router";
import SimpleLayout from "./layouts/SimpleLayout.vue";
import AppLayout from "./layouts/AppLayout.vue";
import { useStore } from "./store";
import { INotificationsSuccess, INotificationsError } from "@/interfaces/INotifications";

export default defineComponent({
  name: "App",
  components: {
    appLayout: AppLayout,
    simpleLayout: SimpleLayout,
  },
  setup() {
    const store = useStore();
    const router = useRouter();

    const layout = computed(() => store.getters["layout/getLayout"]);
    // const token = computed(() => window.location.search.replace("?token=", ""));

    const isLoggedIn = computed(() => store.getters["auth/isLoggedIn"]);
    const hasLoggedID = computed(() => store.getters["auth/id"] !== "");

    onMounted(async () => {
      if (!isLoggedIn.value) {
        store.dispatch("layout/setLayout", "simpleLayout");
      }

      if (!hasLoggedID.value && isLoggedIn.value) {
        try {
          await store.dispatch("auth/logout");

          store.dispatch("layout/setLayout", "simpleLayout");
          router.push("/login");

          store.dispatch(
            "snackbar/showSnackbarSuccessAction",
            INotificationsSuccess.namespaceReload,
          );
        } catch {
          store.dispatch(
            "snackbar/showSnackbarErrorAction",
            INotificationsError.namespaceLoad,
          );
        }
      }
    });

    return {
      layout,
    };
  },
});
</script>
