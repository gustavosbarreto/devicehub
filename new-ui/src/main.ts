import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import router from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

const app = createApp(App);

Sentry.init({
  app,
  dsn: process.env.SHELLHUB_SENTRY_DSN || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  environment: process.env.SHELLHUB_VERSION || "dev",
  release: process.env.SHELLHUB_VERSION || "dev",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(router);
app.use(store, key);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
