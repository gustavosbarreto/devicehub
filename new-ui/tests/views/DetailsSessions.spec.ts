import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import DetailsSessions from "../../src/views/DetailsSessions.vue";
import { createStore } from "vuex";
import { key } from "../../src/store";
import routes from "../../src/router";

describe.skip("DetailsSessions", () => {
  let wrapper: VueWrapper<any>;
  const vuetify = createVuetify();

  const sessionGlobal = {
    uid: "8c354a00",
    device_uid: "a582b47a",
    device: {
      uid: "a582b47a",
      name: "39-5e-2a",
      identity: {
        mac: "00:00:00",
      },
      info: {
        id: "debian",
        pretty_name: "Debian",
        version: "v0.2.5",
      },
      public_key: "xxxxxxxx",
      tenant_id: "00000000",
      last_seen: "2020-05-18T13:27:02.498Z",
      online: true,
      namespace: "user",
    },
    tenant_id: "00000000",
    username: "user",
    ip_address: "00.00.00",
    started_at: "2020-05-18T12:30:28.824Z",
    last_seen: "2020-05-18T12:30:30.205Z",
    active: true,
    authenticated: true,
    recorded: true,
  };

  const tests = [
    {
      description:
        "Session recorded is true and device is online when user has owner role",
      role: {
        type: "owner",
        permission: true,
      },
      variables: {
        session: sessionGlobal,
        enterprise: true,
      },
      data: {
        uid: sessionGlobal.uid,
        session: sessionGlobal,
        dialog: false,
        hide: true,
      },
      computed: {
        isEnterprise: true,
      },
      template: {
        "sessionUid-field": true,
        "sessionUser-field": true,
        "sessionIpAddress-field": true,
        "sessionStartedAt-field": true,
        "sessionLastSeen-field": true,
      },
      templateText: {
        "sessionUid-field": sessionGlobal.uid,
        "sessionUser-field": sessionGlobal.username,
        "sessionIpAddress-field": sessionGlobal.ip_address,
        "sessionStartedAt-field": "Monday, May 18th 2020, 9:30:28 am",
        "sessionLastSeen-field": "Monday, May 18th 2020, 9:30:30 am",
      },
    },
    {
      description:
        "Session recorded is false and device is online when user has owner role",
      role: {
        type: "owner",
        permission: true,
      },
      variables: {
        session: { ...sessionGlobal, recorded: false },
        enterprise: true,
      },
      data: {
        uid: sessionGlobal.uid,
        session: { ...sessionGlobal, recorded: false },
        dialog: false,
        hide: true,
      },
      computed: {
        isEnterprise: true,
      },
      components: {
        "sessionPlay-component": false,
        "sessionClose-component": false,
        "sessionDeleteRecord-component": false,
      },
      template: {
        "sessionUid-field": true,
        "sessionUser-field": true,
        "sessionIpAddress-field": true,
        "sessionStartedAt-field": true,
        "sessionLastSeen-field": true,
      },
      templateText: {
        "sessionUid-field": sessionGlobal.uid,
        "sessionUser-field": sessionGlobal.username,
        "sessionIpAddress-field": sessionGlobal.ip_address,
        "sessionStartedAt-field": "Monday, May 18th 2020, 9:30:28 am",
        "sessionLastSeen-field": "Monday, May 18th 2020, 9:30:30 am",
      },
    },
    {
      description:
        "Session recorded is false and device is offline when user has owner role",
      role: {
        type: "owner",
        permission: true,
      },
      variables: {
        session: {
          ...sessionGlobal,
          device: { online: false },
          active: false,
          recorded: false,
        },
        enterprise: true,
      },
      data: {
        uid: sessionGlobal.uid,
        session: {
          ...sessionGlobal,
          device: { online: false },
          active: false,
          recorded: false,
        },
        dialog: false,
        hide: true,
      },
      computed: {
        isEnterprise: true,
      },
      components: {
        "sessionPlay-component": false,
        "sessionClose-component": false,
        "sessionDeleteRecord-component": false,
      },
      template: {
        "sessionUid-field": true,
        "sessionUser-field": true,
        "sessionIpAddress-field": true,
        "sessionStartedAt-field": true,
        "sessionLastSeen-field": true,
      },
      templateText: {
        "sessionUid-field": sessionGlobal.uid,
        "sessionUser-field": sessionGlobal.username,
        "sessionIpAddress-field": sessionGlobal.ip_address,
        "sessionStartedAt-field": "Monday, May 18th 2020, 9:30:28 am",
        "sessionLastSeen-field": "Monday, May 18th 2020, 9:30:30 am",
      },
    },
    {
      description:
        "Session recorded is true and device is online, but not enterprise",
      role: {
        type: "owner",
        permission: true,
      },
      variables: {
        session: sessionGlobal,
        enterprise: false,
      },
      data: {
        uid: sessionGlobal.uid,
        session: sessionGlobal,
        dialog: false,
        hide: true,
      },
      computed: {
        isEnterprise: false,
      },
      components: {
        "sessionPlay-component": false,
        "sessionClose-component": true,
      },
      template: {
        "sessionUid-field": true,
        "sessionUser-field": true,
        "sessionIpAddress-field": true,
        "sessionStartedAt-field": true,
        "sessionLastSeen-field": true,
      },
      templateText: {
        "sessionUid-field": sessionGlobal.uid,
        "sessionUser-field": sessionGlobal.username,
        "sessionIpAddress-field": sessionGlobal.ip_address,
        "sessionStartedAt-field": "Monday, May 18th 2020, 9:30:28 am",
        "sessionLastSeen-field": "Monday, May 18th 2020, 9:30:30 am",
      },
    },
    {
      description:
        "Session recorded is true and device is online when user has observer role",
      role: {
        type: "observer",
        permission: false,
      },
      variables: {
        session: sessionGlobal,
        enterprise: true,
      },
      data: {
        uid: sessionGlobal.uid,
        session: sessionGlobal,
        dialog: false,
        hide: true,
      },
      computed: {
        isEnterprise: true,
      },
      template: {
        "sessionUid-field": true,
        "sessionUser-field": true,
        "sessionIpAddress-field": true,
        "sessionStartedAt-field": true,
        "sessionLastSeen-field": true,
      },
      templateText: {
        "sessionUid-field": sessionGlobal.uid,
        "sessionUser-field": sessionGlobal.username,
        "sessionIpAddress-field": sessionGlobal.ip_address,
        "sessionStartedAt-field": "Monday, May 18th 2020, 9:30:28 am",
        "sessionLastSeen-field": "Monday, May 18th 2020, 9:30:30 am",
      },
    },
  ];

  const store = (session: any, currentRole: any) => {
    return createStore({
      state: {
        session,
        currentRole,
      },
      getters: {
        "sessions/get": (state) => state.session,
        "auth/role": (state) => state.currentRole,
      },
      actions: {
        "sessions/get": vi.fn(),
        "sessions/close": vi.fn(),
        "snackbar/showSnackbarErrorLoading": vi.fn(),
      },
    });
  };

  tests.forEach((test) => {
    describe(`${test.description}`, () => {
      beforeEach(async () => {
        wrapper = mount(DetailsSessions, {
          global: {
            plugins: [
              [store(test.variables.session, test.role.type), key],
              vuetify,
              routes,
            ],
          },
        });
      });

      ///////
      // Component Rendering
      //////

      it("Is a Vue instance", () => {
        expect(wrapper).toBeTruthy();
      });
      it("Renders the component", () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

      ///////
      // Data checking
      //////
      it("Data is defined", () => {
        expect(wrapper.vm.$data).toBeDefined();
      });
      it('Process data in the computed', () => {
        expect(wrapper.vm.session).toEqual(test.data.session);
      });

      //////
      // HTML validation
      //////

      it("Renders the template with data", () => {
        for (const [key, value] of Object.entries(test.template)) {
          expect(wrapper.find(`[data-test="${key}"]`).exists()).toBe(value);
        }
      });
      it("Renders template with expected text", () => {
        for (const [key, value] of Object.entries(test.templateText)) {
          expect(wrapper.find(`[data-test="${key}"]`).text()).toBe(value);
        }
      });
    });
  });
});
