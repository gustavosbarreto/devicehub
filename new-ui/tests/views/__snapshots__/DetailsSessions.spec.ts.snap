// Vitest Snapshot v1

exports[`DetailsSessions > Session recorded is false and device is offline when user has owner role > Renders the component 1`] = `
"<div class=\\"d-flex pa-0 align-center\\">
  <h1>Session Details</h1>
</div>
<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2 bg-v-theme-surface\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title pa-4 d-flex align-center justify-space-between\\">
    <div class=\\"d-flex align-center\\"><i class=\\"mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-small\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-0\\"></i>
      <!--teleport start-->
      <!--teleport end--><span class=\\"ml-2\\"></span>
    </div>
    <div><span class=\\"v-chip v-chip--link v-theme--light v-chip--density-comfortable v-chip--size-small v-chip--variant-tonal\\" draggable=\\"false\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-chip__overlay\\"></span><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!----><i class=\\"mdi-dots-horizontal mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
      <!---->
      <!----></span>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-card-text pa-4 pt-0\\">
    <div>
      <div class=\\"text-overline mt-3\\">uid:</div>
      <div data-test=\\"sessionUid-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">user:</div>
      <div data-test=\\"sessionUser-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Authenticated:</div>
      <div data-test=\\"sessionAuthenticated-field\\"><i class=\\"mdi-shield-alert mdi v-icon notranslate v-theme--light v-icon--size-default text-error\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-3\\"></i>
        <!---->
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">IP address:</div>
      <div data-test=\\"sessionIpAddress-field\\"><code class=\\"bg-tabs pa-1\\"></code></div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Started:</div>
      <div data-test=\\"sessionStartedAt-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div>Last seen:</div>
      <div data-test=\\"sessionLastSeen-field\\">
        <p></p>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;

exports[`DetailsSessions > Session recorded is false and device is online when user has owner role > Renders the component 1`] = `
"<div class=\\"d-flex pa-0 align-center\\">
  <h1>Session Details</h1>
</div>
<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2 bg-v-theme-surface\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title pa-4 d-flex align-center justify-space-between\\">
    <div class=\\"d-flex align-center\\"><i class=\\"mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-small text-success\\" aria-hidden=\\"true\\"></i><span class=\\"ml-2\\">39-5e-2a</span></div>
    <div><span class=\\"v-chip v-chip--link v-theme--light v-chip--density-comfortable v-chip--size-small v-chip--variant-tonal\\" draggable=\\"false\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-chip__overlay\\"></span><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!----><i class=\\"mdi-dots-horizontal mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
      <!---->
      <!----></span>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-card-text pa-4 pt-0\\">
    <div>
      <div class=\\"text-overline mt-3\\">uid:</div>
      <div data-test=\\"sessionUid-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">user:</div>
      <div data-test=\\"sessionUser-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Authenticated:</div>
      <div data-test=\\"sessionAuthenticated-field\\"><i class=\\"mdi-shield-alert mdi v-icon notranslate v-theme--light v-icon--size-default text-error\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-3\\"></i>
        <!---->
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">IP address:</div>
      <div data-test=\\"sessionIpAddress-field\\"><code class=\\"bg-tabs pa-1\\"></code></div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Started:</div>
      <div data-test=\\"sessionStartedAt-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div>Last seen:</div>
      <div data-test=\\"sessionLastSeen-field\\">
        <p></p>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;

exports[`DetailsSessions > Session recorded is true and device is online when user has observer role > Renders the component 1`] = `
"<div class=\\"d-flex pa-0 align-center\\">
  <h1>Session Details</h1>
</div>
<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2 bg-v-theme-surface\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title pa-4 d-flex align-center justify-space-between\\">
    <div class=\\"d-flex align-center\\"><i class=\\"mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-small text-success\\" aria-hidden=\\"true\\"></i><span class=\\"ml-2\\">39-5e-2a</span></div>
    <div><span class=\\"v-chip v-chip--link v-theme--light v-chip--density-comfortable v-chip--size-small v-chip--variant-tonal\\" draggable=\\"false\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-chip__overlay\\"></span><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!----><i class=\\"mdi-dots-horizontal mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
      <!---->
      <!----></span>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-card-text pa-4 pt-0\\">
    <div>
      <div class=\\"text-overline mt-3\\">uid:</div>
      <div data-test=\\"sessionUid-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">user:</div>
      <div data-test=\\"sessionUser-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Authenticated:</div>
      <div data-test=\\"sessionAuthenticated-field\\"><i class=\\"mdi-shield-alert mdi v-icon notranslate v-theme--light v-icon--size-default text-error\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-3\\"></i>
        <!---->
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">IP address:</div>
      <div data-test=\\"sessionIpAddress-field\\"><code class=\\"bg-tabs pa-1\\"></code></div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Started:</div>
      <div data-test=\\"sessionStartedAt-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div>Last seen:</div>
      <div data-test=\\"sessionLastSeen-field\\">
        <p></p>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;

exports[`DetailsSessions > Session recorded is true and device is online when user has owner role > Renders the component 1`] = `
"<div class=\\"d-flex pa-0 align-center\\">
  <h1>Session Details</h1>
</div>
<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2 bg-v-theme-surface\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title pa-4 d-flex align-center justify-space-between\\">
    <div class=\\"d-flex align-center\\"><i class=\\"mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-small text-success\\" aria-hidden=\\"true\\"></i><span class=\\"ml-2\\">39-5e-2a</span></div>
    <div><span class=\\"v-chip v-chip--link v-theme--light v-chip--density-comfortable v-chip--size-small v-chip--variant-tonal\\" draggable=\\"false\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-chip__overlay\\"></span><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!----><i class=\\"mdi-dots-horizontal mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
      <!---->
      <!----></span>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-card-text pa-4 pt-0\\">
    <div>
      <div class=\\"text-overline mt-3\\">uid:</div>
      <div data-test=\\"sessionUid-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">user:</div>
      <div data-test=\\"sessionUser-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Authenticated:</div>
      <div data-test=\\"sessionAuthenticated-field\\"><i class=\\"mdi-shield-alert mdi v-icon notranslate v-theme--light v-icon--size-default text-error\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-3\\"></i>
        <!---->
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">IP address:</div>
      <div data-test=\\"sessionIpAddress-field\\"><code class=\\"bg-tabs pa-1\\"></code></div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Started:</div>
      <div data-test=\\"sessionStartedAt-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div>Last seen:</div>
      <div data-test=\\"sessionLastSeen-field\\">
        <p></p>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;

exports[`DetailsSessions > Session recorded is true and device is online, but not enterprise > Renders the component 1`] = `
"<div class=\\"d-flex pa-0 align-center\\">
  <h1>Session Details</h1>
</div>
<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2 bg-v-theme-surface\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title pa-4 d-flex align-center justify-space-between\\">
    <div class=\\"d-flex align-center\\"><i class=\\"mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-small text-success\\" aria-hidden=\\"true\\"></i><span class=\\"ml-2\\">39-5e-2a</span></div>
    <div><span class=\\"v-chip v-chip--link v-theme--light v-chip--density-comfortable v-chip--size-small v-chip--variant-tonal\\" draggable=\\"false\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-chip__overlay\\"></span><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!----><i class=\\"mdi-dots-horizontal mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i>
      <!---->
      <!----></span>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <div class=\\"v-card-text pa-4 pt-0\\">
    <div>
      <div class=\\"text-overline mt-3\\">uid:</div>
      <div data-test=\\"sessionUid-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">user:</div>
      <div data-test=\\"sessionUser-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Authenticated:</div>
      <div data-test=\\"sessionAuthenticated-field\\"><i class=\\"mdi-shield-alert mdi v-icon notranslate v-theme--light v-icon--size-default text-error\\" aria-hidden=\\"true\\" aria-describedby=\\"v-tooltip-3\\"></i>
        <!---->
      </div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">IP address:</div>
      <div data-test=\\"sessionIpAddress-field\\"><code class=\\"bg-tabs pa-1\\"></code></div>
    </div>
    <div>
      <div class=\\"text-overline mt-3\\">Started:</div>
      <div data-test=\\"sessionStartedAt-field\\">
        <p></p>
      </div>
    </div>
    <div>
      <div>Last seen:</div>
      <div data-test=\\"sessionLastSeen-field\\">
        <p></p>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;
